### PR TITLE
Service Bus Activities enhancements - Workflow Inbox and Test

### PR DESCRIPTION
**Pull request by jdevillard**
This PR add integration testing for Azure Service Bus Activities. 
This PR also modify the Azure Service Bus activities to use the WorkflowInbox

### CODE REVIEW

**Review by sfmskywalker — APPROVED** [review]

**Comment by jdevillard** [comment]
I don't know why all the GUID in the SLN have change ? maybe a mismatch version of my VS2022?

**Comment by sfmskywalker** [comment]
Indeed, quite strange that it would do that. But as long as the projects and their references remain in tact, I don't think it's an issue.

File: test/integration/Elsa.ServiceBusIntegrationTests/Elsa.ServiceBusIntegrationTests.csproj [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.5">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="coverlet.collector" Version="3.2.0">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+
+	  <PackageReference Include="Moq" Version="4.18.4" />
```
Comment (sfmskywalker) [inline test/integration/Elsa.ServiceBusIntegrationTests/Elsa.ServiceBusIntegrationTests.csproj:24]: Ah, sorry I overlooked this reference - [we no longer use Moq](https://www.youtube.com/watch?v=A06nNjBKV7I). Instead, NSubstitute seems to be a good alternative. There's also FakeItEasy, but hasn't as much traction it seems (not sure why).

**Review by sfmskywalker — CHANGES_REQUESTED** [review]
Please replace Moq with another mocking package, such as NSubstitute or FakeItEasy.

**Comment by jdevillard** [comment]
ok SLN conflict is resolved ! if ok for you for the guid change in Sln , seems to be good and this could be merge.

**Pull request merged**